tools/realsense-viewer/realsense-viewer.cpp

```diff
@@ -130,18 +131,43 @@ void add_playback_device( context & ctx,
 // This function is called every frame
 // If between the frames there was an asyncronous connect/disconnect event
 // the function will pick up on this and add the device to the viewer
-bool refresh_devices(std::mutex& m,
+void refresh_devices(std::mutex& m,
     context& ctx,
     device_changes& devices_connection_changes,
     std::vector<device>& current_connected_devices,
+    std::vector<device>& future_initialization_devices,
+    bool allow_future_initialization,
     std::vector<std::pair<std::string, std::string>>& device_names,
     device_models_list& device_models,
     viewer_model& viewer_model,
     std::string& error_message)
 {
+    
     event_information info({}, {});
     if (!devices_connection_changes.try_get_next_changes(info))
-        return false;
+    {
+        if (allow_future_initialization)
+        {
+            auto dev_itr = begin(future_initialization_devices);
+            while (dev_itr != end(future_initialization_devices))
```

Comment:
Style only - I would discard the flag and use the vector as indication if a job is needed.

while( ! future_initialization_devices.empty() )
{
    //Handle a device
    //Erase that device from the list
}